### PR TITLE
[NDM] Add Cloud application metrics query to the Cisco SD-WAN client

### PR DESCRIPTION
**Pull request by TCheruy**
### What does this PR do?
Add ability to query Cloud application metrics in the Cisco SD-WAN client

### CODE REVIEW

File: pkg/collector/corechecks/network-devices/cisco-sdwan/client/client_test.go [hunk 1]
```diff
@@ -538,3 +538,52 @@ func TestGetHardwareStates(t *testing.T) {
...
+	require.Equal(t, "amazon_aws", devices[0].Application)
+	require.Equal(t, "amazon-group", devices[0].NbarAppGroupName)
+	require.Equal(t, float64(1), devices[0].VpnID)
+	require.Equal(t, float64(1721819993833), devices[0].EntryTime)
+	require.Equal(t, float64(404), devices[0].Latency)
+	require.Equal(t, float64(0), devices[0].Loss)
+	require.Equal(t, "5.0", devices[0].VqeScore)
+
+	// Ensure endpoint has been called 1 times
+	require.Equal(t, 1, handler.numberOfCalls())
```
Comment (NouhaManai96) [inline pkg/collector/corechecks/network-devices/cisco-sdwan/client/client_test.go:588]: in which cases the endpoint would be called multiple times ?

**Review by NouhaManai96 — APPROVED** [review]

File: pkg/collector/corechecks/network-devices/cisco-sdwan/client/client_test.go (hunk 1 above)
Comment (TCheruy) [reply pkg/collector/corechecks/network-devices/cisco-sdwan/client/client_test.go:588]: Pagination, if for example we need to query more data points than the `maxCount` param, the client will make multiple queries to get everything

Comment (NouhaManai96) [reply pkg/collector/corechecks/network-devices/cisco-sdwan/client/client_test.go:588]: I see 👍 
should there be a test case for that ?

Comment (TCheruy) [reply pkg/collector/corechecks/network-devices/cisco-sdwan/client/client_test.go:588]: The pagination logic is shared between endpoints so I didn't write pagination tests for each specific endpoint, but common pagination tests are here: https://github.com/DataDog/datadog-agent/blob/def18e3815c6cc67d8669408032961804821687a/pkg/collector/corechecks/network-devices/cisco-sdwan/client/request_test.go#L361-L422

**Pull request merged**